cmd/kubeadm/app/discovery/file/file.go

```diff
@@ -116,14 +152,30 @@ func tryParseClusterInfoFromConfigMap(cm *v1.ConfigMap) (*clientcmdapi.Config, e
 	return parsedKubeConfig, nil
 }
 
-// validateClusterInfoKubeConfig makes sure the user-provided cluster-info KubeConfig file is valid
-func validateClusterInfoKubeConfig(clusterinfo *clientcmdapi.Config) error {
-	if len(clusterinfo.Clusters) < 1 {
+// validateKubeConfig makes sure the user-provided KubeConfig file is valid
+func validateKubeConfig(config *clientcmdapi.Config) error {
+	if len(config.Clusters) < 1 {
 		return fmt.Errorf("the provided cluster-info KubeConfig file must have at least one Cluster defined")
 	}
-	defaultCluster := kubeconfigutil.GetClusterFromKubeConfig(clusterinfo)
+	defaultCluster := kubeconfigutil.GetClusterFromKubeConfig(config)
 	if defaultCluster == nil {
 		return fmt.Errorf("the provided cluster-info KubeConfig file must have an unnamed Cluster or a CurrentContext that specifies a non-nil Cluster")
 	}
-	return nil
+	return clientcmd.Validate(*config)
+}
+
+// extractFromFile tries to read the pem file
+func extractFromFile(filename string) ([]byte, error) {
+	glog.V(3).Infof("Reading config file %q", filename)
```

Comment:
don't we have the generic "open file" -> return `[]byte, error` function already?
if not, i think this should go in utils - e.g. `readFileBytes()`?

this would avoid the imports here too:
```
+	"io/ioutil"
+	"os"
+
+	"github.com/golang/glog"
```